include/config.py

```diff
@@ -43,18 +45,18 @@
 
 # The idea is to have different ADC resolution value for each ADC.
 # The number of ADC if defined by num_adc property. Currently it is 2 * num_matrix(2) = 4
-# NOTE: Only taking in account indexes 0 and 2, 1 and 3 are ignored, because ADCs 1 and 3 are assumed t be equal to 0 and 2. 
+# NOTE: For latency computation only taking in account indexes 0 and 2, 1 and 3 are ignored, because ADCs 1 and 3 are assumed to be equal to 0 and 2.
 adc_res_new = {
                 'matrix_adc_0' : 8,
-                'matrix_adc_1' : 4,
+                'matrix_adc_1' : 8,
                 'matrix_adc_2' : 8,
-                'matrix_adc_3' : 4
+                'matrix_adc_3' : 8
```

Comment:
For lines 49-53, I understand this would only work for configuration with `num_matrix=2`. Can you either put this within a new flag (maybe `if hetero-adc`), so that the `adc_res_new` dictionary is only created when `flag==true`, otherwise it just creates an empty `adc_res_new` dictionary. This will ensure that if `num_matrix>2`, `ima.py::init` will just fall back to using old `adc_res` for all ADCs.

Reply:
As I changed the way that other dictionaries will be access because it will depend on the value of adc_res_new (we can change that for a better name) I think what we could do is is create a new flag as you suggested, hetero-adc, but uses the old adc_res as default value for adc_res_new to avoid change the other dictionaries again. Something like:
hetero-adc = false
if not(hetero-adc): 
    adc_res_new = {
                'matrix_adc_0' : adc_res,
                'matrix_adc_1' : adc_res,
                'matrix_adc_2' : adc_res,
                'matrix_adc_3' : adc_res
    }
else:
    adc_res_new = {
                'matrix_adc_0' : 4,
                'matrix_adc_1' : 6
                'matrix_adc_2' : 4
                'matrix_adc_3' : 6
    }
What do you think?